### PR TITLE
Forman 45 revise gen

### LINKED ISSUE #45: xcube gen generates duplicates in time and unsorted time dimension for S2+
**Describe the bug**
xcube gen generates two identical time slices with S2+ plugin. In the input data there is only one input file. In addition, the time dimension is not sorted. 

**To Reproduce**
The cube has been produced in xcube-gen with the the following command. 
nohup xcube gen --append -v CHL_GILERSON2010_GLOBAL,KdPAR,SPM_VITOnir_SCHELDT,TUR_NECHAD2009_GLOBAL_665 -c /home/xcube/projects/xcube-services/dcs4cop/xcube-gen-configs/dcs4cop-gen_BC_config_S2.yml /data/EOdata/related/DCS4COP/cube_input/VITO/2017/*/*.nc > nohup.out &
The resulting zarr file is still on xcube-gen at /home/xcube/xcube-output/ dcs4cop-bc-s2-sns-l2c-v1.zarr
Note that this issue might be related to the fact that the generation has been interrupted and started again with the same output name. 

**Screenshots**
image

### PR DESCRIPTION
**Pull request by forman**
Closes #45

Also reworked `xcube gen` to be more user-friendly and more consistent with other tools. 
The changes are
- Removed `--dir` and `--name` options and replaced it by single `--output` option, 
whose default value is `out.zarr`. (#45)
- The `--format` option no longer has a default value. If not given, 
format is guessed from `--output` option.
- Renamed following parameters in the configuration file:
+ `input_files` into `input_paths`, also because paths may point into object storage 
  locations (buckets);  
+ `output_file` into `output_path`, to be consistent with `input_paths`.  

While implementing this PR and after implementing `xcube resample`, I recognized there is still a lot inconsistency across the tools. I'll create a new ticket for this.

### CODE REVIEW

**Comment by forman** [comment]
@AliceBalfanz and @dzelge could you please complete your PR until tomorrow evening, because I'd like to demo and use the updated `xcube gen` on Tuesday.

@TonioF, FYI.

**Comment by forman** [comment]
Partly addresses #91.

File: README.md [hunk 1]
```diff
@@ -441,13 +441,9 @@ handy tool [Wicket](https://arthur-e.github.io/Wicket/sandbox-gmaps3.html).
...
     
 
-
-
-
-
 Example:
 
-    $ xcube gen -a -s 2000,1000 -r 0,50,5,52.5 -v conc_chl,conc_tsm,kd489,c2rcc_flags,quality_flags -n hiroc-cube -t snap-c2rcc D:\OneDrive\BC\EOData\HIGHROC\2017\01\*.nc
+    $ xcube gen -p snap-olci-highroc-l2 -a -s 2000,1000 -r 0,50,5,52.5 -v conc_chl,conc_tsm,kd489,c2rcc_flags,quality_flags -o hiroc-cube D:\OneDrive\BC\EOData\HIGHROC\2017\**\*.nc
```
Comment (AliceBalfanz) [inline README.md]:
```suggestion
    $ xcube gen -p snap-olci-highroc-l2 -a -s 2000,1000 -r 0,50,5,52.5 -v conc_chl,conc_tsm,kd489,c2rcc_flags,quality_flags -o hiroc-cube.zarr D:\OneDrive\BC\EOData\HIGHROC\2017\**\*.nc
```

File: xcube/api/gen/config.py [hunk 2]
```diff
@@ -47,58 +46,50 @@ def get_config_dict(config_obj: Dict[str, Union[str, bool, int, float, list, dic
...
         config['output_region'] = output_region
 
     if output_variables is not None:
         output_variables = list(map(lambda c: c.strip(), output_variables.split(',')))
-        if output_variables == ['']:
-            raise ValueError('output_variables must contain at least one name')
-        if any([var_name == '' for var_name in output_variables]):
-            raise ValueError('all names in output_variables must be non-empty')
+        if output_variables == [''] or any([var_name == '' for var_name in output_variables]):
+            raise ValueError('output_variables must be a list of existing variable names')
```
Comment (AliceBalfanz) [inline xcube/api/gen/config.py]:
```suggestion
            raise ValueError('Output_variables must be a list of existing variable names')
```

**Comment by dzelge** [comment]
Bin dabei. 

 

Von: Norman Fomferra <notifications@github.com> 
Gesendet: Sonntag, 16. Juni 2019 19:06
An: dcs4cop/xcube <xcube@noreply.github.com>
Cc: Helge Dzierzon <helge.dzierzon@brockmann-consult.de>; Mention <mention@noreply.github.com>
Betreff: Re: [dcs4cop/xcube] Forman 45 revise gen (#90)

 

@AliceBalfanz <https://github.com/AliceBalfanz>  and @dzelge <https://github.com/dzelge>  could you please complete your PR until tomorrow evening, because I'd like to demo and use the updated xcube gen on Tuesday.

@TonioF <https://github.com/TonioF> , FYI.

**Comment by forman** [comment]
Merging as I need it tomorrow...

**Pull request closed without merging**